agent/status_test.go

```diff
@@ -73,3 +73,30 @@ func (s *StatusSuite) TestAgentStartsStatusServer() {
 	s.Require().NoError(err)
 	s.Equal(200, resp.StatusCode)
 }
+
+func (s *StatusSuite) TestAgentFailsToStartTwice() {
+	resp, err := http.Get("http://127.0.0.1:2287/status")
+	s.Error(err)
+
+	s.testOpts.StatusPort = 2287
+	agt := New(s.testOpts, client.NewMock("url"))
+	mockCommunicator := agt.comm.(*client.Mock)
+	mockCommunicator.NextTaskIsNil = true
+	ctx, cancel := context.WithCancel(context.Background())
+	s.cancel = cancel
+
+	go func() {
+		err = agt.Start(ctx)
+		s.NoError(err)
+	}()
+	time.Sleep(100 * time.Millisecond)
+	resp, err = http.Get("http://127.0.0.1:2287/status")
+	s.Require().NoError(err)
```

Comment:
maybe we should retry getting the status in a loop so if it takes more than 100ms to start the first server the test won't fail?